src/main.rs

```diff
@@ -31,11 +40,65 @@ impl BFInterp {
         }
     }
 
+    fn store_gl_var(&mut self, module: &Module) {
+        for gl_var in module.global_vars.iter() {
+            match &gl_var.initializer {
+                Some(init) => {
+                    self.gl_vars
+                        .insert(gl_var.name.clone(), init.as_ref().clone());
+                }
+                None => todo!(),
+            }
+        }
+    }
+
     fn it_bb(&mut self, bb: &BasicBlock) -> Option<u64> {
         for inst in bb.instrs.iter() {
             match inst {
                 Instruction::Alloca(_) => {}
                 Instruction::Store(store) => self.store_var(&store.address, &store.value),
+                Instruction::Load(load) => self.load(&load.address, &load.dest),
+                Instruction::Call(call) => self.call_func(call),
+                Instruction::Add(add) => {
+                    self.int_bin_operations(&add.operand0, &add.operand1, &add.dest, 0)
```

Comment:
I think we should create an enum here instead of passing hard coded integers to tell which operation we are using.